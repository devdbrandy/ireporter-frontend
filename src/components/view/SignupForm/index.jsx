import React from 'react';

const SignupForm = () => (
  <form className="signup--form" id="signup-form">
    <p className="hint">Fill the form below to create an account</p>
    <hr className="dash" />
    <div className="input-wrapper">
      <input type="text" name="firstname" placeholder="First name" minLength="3" required />
    </div>
    <div className="input-wrapper">
      <input type="text" name="lastname" placeholder="Last name" minLength="3" required />
    </div>
    <div className="input-wrapper">
      <input
        type="text"
        name="phoneNumber"
        id="phonenumber"
        placeholder="Phone number"
        minLength="11"
        required
      />
    </div>
    <div className="input-wrapper">
      <input
        type="email"
        name="email"
        placeholder="Email address"
        pattern="^([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x22([^\x0d\x22\x5c\x80-\xff]|\x5c[\x00-\x7f])*\x22)(\x2e([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x22([^\x0d\x22\x5c\x80-\xff]|\x5c[\x00-\x7f])*\x22))*\x40([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x5b([^\x0d\x5b-\x5d\x80-\xff]|\x5c[\x00-\x7f])*\x5d)(\x2e([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x5b([^\x0d\x5b-\x5d\x80-\xff]|\x5c[\x00-\x7f])*\x5d))*$"
      />
    </div>
    <div className="input-wrapper">
      <input type="text" name="username" placeholder="Username" required />
    </div>
    <div className="input-wrapper">
      <input
        type="password"
        name="password"
        placeholder="Password"
        pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
        title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
        minLength="6"
        required
      />
      <i className="fas fa-eye icon toggle-pwd" />
    </div>
    <div className="input-wrapper">
      <input type="password" name="passwordConfirmation" placeholder="Repeat Password" required />
      <i className="fas fa-eye icon toggle-pwd" />
    </div>
    <button type="submit" className="btn btn-primary mb-10">Register</button>
  </form>
);

export default SignupForm;
