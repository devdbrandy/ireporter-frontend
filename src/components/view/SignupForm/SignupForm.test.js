import React from 'react';
import { shallow } from 'enzyme';
import SignupForm from './index';

describe('<SignupForm />', () => {
  it('renders without crashing given the required props', () => {
    const props = {
      handleSignUp: jest.fn()
    };
    const enzymeWrapper = shallow(<SignupForm {...props} />);
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
