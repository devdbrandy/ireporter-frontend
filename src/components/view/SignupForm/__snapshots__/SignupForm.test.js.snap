// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupForm /> renders without crashing given the required props 1`] = `
<form
  className="signup--form"
  id="signup-form"
  onSubmit={[Function]}
>
  <p
    className="hint"
  >
    Fill the form below to create an account
  </p>
  <hr
    className="dash"
  />
  <div
    className="input-wrapper"
  >
    <input
      minLength="3"
      name="firstname"
      onChange={[Function]}
      placeholder="First name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      minLength="3"
      name="lastname"
      onChange={[Function]}
      placeholder="Last name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      id="phonenumber"
      minLength="11"
      name="phoneNumber"
      onChange={[Function]}
      placeholder="Phone number"
      required={true}
      type="text"
    />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      name="email"
      onChange={[Function]}
      pattern="^([^\\\\x00-\\\\x20\\\\x22\\\\x28\\\\x29\\\\x2c\\\\x2e\\\\x3a-\\\\x3c\\\\x3e\\\\x40\\\\x5b-\\\\x5d\\\\x7f-\\\\xff]+|\\\\x22([^\\\\x0d\\\\x22\\\\x5c\\\\x80-\\\\xff]|\\\\x5c[\\\\x00-\\\\x7f])*\\\\x22)(\\\\x2e([^\\\\x00-\\\\x20\\\\x22\\\\x28\\\\x29\\\\x2c\\\\x2e\\\\x3a-\\\\x3c\\\\x3e\\\\x40\\\\x5b-\\\\x5d\\\\x7f-\\\\xff]+|\\\\x22([^\\\\x0d\\\\x22\\\\x5c\\\\x80-\\\\xff]|\\\\x5c[\\\\x00-\\\\x7f])*\\\\x22))*\\\\x40([^\\\\x00-\\\\x20\\\\x22\\\\x28\\\\x29\\\\x2c\\\\x2e\\\\x3a-\\\\x3c\\\\x3e\\\\x40\\\\x5b-\\\\x5d\\\\x7f-\\\\xff]+|\\\\x5b([^\\\\x0d\\\\x5b-\\\\x5d\\\\x80-\\\\xff]|\\\\x5c[\\\\x00-\\\\x7f])*\\\\x5d)(\\\\x2e([^\\\\x00-\\\\x20\\\\x22\\\\x28\\\\x29\\\\x2c\\\\x2e\\\\x3a-\\\\x3c\\\\x3e\\\\x40\\\\x5b-\\\\x5d\\\\x7f-\\\\xff]+|\\\\x5b([^\\\\x0d\\\\x5b-\\\\x5d\\\\x80-\\\\xff]|\\\\x5c[\\\\x00-\\\\x7f])*\\\\x5d))*$"
      placeholder="Email address"
      type="email"
    />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      name="username"
      onChange={[Function]}
      placeholder="Username"
      required={true}
      type="text"
    />
  </div>
  <PasswordInput
    handleOnChange={[Function]}
    label=""
    labelText=""
    minLength="6"
    name="password"
    pattern="(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
    placeholder="Password"
    title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
  />
  <PasswordInput
    handleOnChange={[Function]}
    label=""
    labelText=""
    minLength={null}
    name="passwordConfirmation"
    pattern={null}
    placeholder="Repeat password"
    title={null}
  />
  <button
    className="btn btn-primary mb-10"
    type="submit"
  >
    Register
  </button>
</form>
`;
